src/TelemetryChannels/ServerTelemetryChannel/Shared/TelemetryChannelBuilderExtensions.cs

```diff
@@ -42,7 +90,33 @@ public static TelemetryProcessorChainBuilder UseAdaptiveSampling(this TelemetryP
         /// <return>Instance of <see cref="TelemetryProcessorChainBuilder"/>.</return>
         public static TelemetryProcessorChainBuilder UseAdaptiveSampling(this TelemetryProcessorChainBuilder builder, double maxTelemetryItemsPerSecond)
         {
-            return builder.Use((next) => new AdaptiveSamplingTelemetryProcessor(next) { MaxTelemetryItemsPerSecond = maxTelemetryItemsPerSecond });
+            if (builder == null)
+            {
+                throw new ArgumentNullException("builder");
+            }
+
+            return builder.Use(next => new AdaptiveSamplingTelemetryProcessor(next) { MaxTelemetryItemsPerSecond = maxTelemetryItemsPerSecond });
+        }
+
+        /// <summary>
+        /// Adds <see cref="AdaptiveSamplingTelemetryProcessor"/> to the <see cref="TelemetryProcessorChainBuilder" />
+        /// </summary>
+        /// <param name="builder">Instance of <see cref="TelemetryProcessorChainBuilder"/></param>
+        /// <param name="maxTelemetryItemsPerSecond">Maximum number of telemetry items to be generated on this application instance.</param>
+        /// <param name="excludedTypes">Semicolon separated list of types that should not be sampled.</param>
+        /// <return>Instance of <see cref="TelemetryProcessorChainBuilder"/>.</return>
+        public static TelemetryProcessorChainBuilder UseAdaptiveSampling(this TelemetryProcessorChainBuilder builder, double maxTelemetryItemsPerSecond, string excludedTypes)
```

Comment:
Given the number of overloads needing to be added it might be time to consider having a single signature with a simple config object (e.g. property bag).